.github/workflows/docker.yml

```diff
@@ -0,0 +1,24 @@
+name: docker build and push
+on:
+  pull_request:
```

Comment:
I believe it would be more efficient to run this workflow for every push instead of every pull request a user raises.

Reply:
totally agree